arrow-libs/core/arrow-autoclose/src/jvmMain/kotlin/arrow/throwIfFatal.kt

```diff
@@ -1,10 +1,8 @@
 package arrow
 
-import kotlin.coroutines.cancellation.CancellationException
-
 @PublishedApi
 internal actual fun Throwable.throwIfFatal(): Throwable =
   when(this) {
-    is VirtualMachineError, is ThreadDeath, is InterruptedException, is LinkageError, is CancellationException -> throw this
+    is VirtualMachineError, is ThreadDeath, is InterruptedException, is LinkageError -> throw this
```

Comment:
Hate that this function cannot be private. I really do not want to expose it 😓

This is why I had the `CancellationException` explicitly in the try/catch, because I rather had a correct definition here, than one specific to AutoCloseable...